oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/strategy/AssignDataCenterToTemplate.java

```
@@ -0,0 +1,89 @@
+import org.jclouds.compute.strategy.impl.CreateNodesWithGroupEncodedIntoNameThenAddToSet;
+import org.jclouds.logging.Logger;
+
+@Beta
+@Singleton
+public class AssignDataCenterToTemplate extends CreateNodesWithGroupEncodedIntoNameThenAddToSet {
+
+   @Resource
+   @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+   protected Logger logger = Logger.NULL;
+
+   private final OneAndOneApi api;
+
+   @Inject
+   protected AssignDataCenterToTemplate(
+           CreateNodeWithGroupEncodedIntoName addNodeWithGroupStrategy,
+           ListNodesStrategy listNodesStrategy,
+           GroupNamingConvention.Factory namingConvention,
+           @Named(PROPERTY_USER_THREADS) ListeningExecutorService userExecutor,
+           CustomizeNodeAndAddToGoodMapOrPutExceptionIntoBadMap.Factory customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory,
+           OneAndOneApi api) {
+      super(addNodeWithGroupStrategy, listNodesStrategy, namingConvention, userExecutor,
+              customizeNodeAndAddToGoodMapOrPutExceptionIntoBadMapFactory);
+      this.api = api;
+   }
+
+   @Override
+   public Map<?, ListenableFuture<Void>> execute(String group, int count, final Template template,
+           Set<NodeMetadata> goodNodes, Map<NodeMetadata, Exception> badNodes,
+           Multimap<NodeMetadata, CustomizationResponse> customizationResponses) {
+
+      logger.info(">> looking for a datacenter in %s", template.getLocation().getId());
+
+      // Try to find an existing datacenter in the selected location
+      DataCenter dataCenter = find(api.dataCenterApi().list(), new Predicate<DataCenter>() {
+         @Override
+         public boolean apply(DataCenter input) {
+            // The location field is not populated when getting the list of datacenters
+            DataCenter details = api.dataCenterApi().get(input.id());
+            return details != null && template.getLocation().getId().equals((details.countryCode().toLowerCase()));
+         }
+      }, null);
```

Don't use `null` as a default value and let it fail, since it is mandatory.